### PR TITLE
Feature/backend profile

### PR DESCRIPTION
**Pull request by APeev89**
Create logig for change password and change email

### CODE REVIEW

File: CryptoWorld.News/CryptoWorld.News.Core/Services/AccountService.cs [hunk 1]
```diff
@@ -16,16 +20,26 @@ namespace CryptoWorld.News.Core.Services
...
     {
         private readonly UserManager<ApplicationUser> userManager;
+        private readonly IConfiguration config;
+        private string secretKey;
         private readonly SignInManager<ApplicationUser> signInManager;
         private readonly IEmailSenderService emailSenderService;
+
         public AccountService(
-            UserManager<ApplicationUser> _userManager, SignInManager<ApplicationUser> _signInManager, IEmailSenderService _emailSenderService)
+                UserManager<ApplicationUser> _userManager,
```
Comment (VelinArbov) [inline CryptoWorld.News/CryptoWorld.News.Core/Services/AccountService.cs]: In the constructor they should be without _ conventions is _ for private fields in our case should be private string _secretKey. Please fix it in all files.

File: CryptoWorld.News/CryptoWorld.News.Core/Services/AccountService.cs [hunk 2]
```diff
@@ -142,8 +156,8 @@ private string GenerateJwtToken(ApplicationUser user)
                 new Claim(ClaimTypes.Name, user.UserName),
                 new Claim(ClaimTypes.Email, user.Email)
             };
-            var secretKey = "abcdefghijklmnopqrstuvwxyzabcdef";
-            var key = new SymmetricSecurityKey(Encoding.UTF8.GetBytes(secretKey));
+
+            var key = new SymmetricSecurityKey(Encoding.UTF8.GetBytes(this.secretKey));
```
Comment (VelinArbov) [inline CryptoWorld.News/CryptoWorld.News.Core/Services/AccountService.cs]: here should be _secretKey

File: CryptoWorld.News/CryptoWorld.News.Core/ViewModels/ChangeEmailModel.cs [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+using System.Threading.Tasks;
+
+namespace CryptоWorld.News.Core.ViewModels
+{
+    public class ChangeEmailModel
+    {
+        [Required]
+        [EmailAddress]
+        public string CurrentEmail { get; set; }
+
```
Comment (VelinArbov) [inline CryptoWorld.News/CryptoWorld.News.Core/ViewModels/ChangeEmailModel.cs]: empty space

File: CryptoWorld.News/CryptoWorld.News.Server/Controllers/UserProfileController.cs [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+                return BadRequest(ModelState);
+            }
+
+            return Ok(new { Message = "The user email was changed successfully!" });
+        }
+
+        [HttpPost("changePassword")]
+        public async Task<IActionResult> ChangePassword([FromBody] ChangePasswordModel model)
+        {
+
```
Comment (VelinArbov) [inline CryptoWorld.News/CryptoWorld.News.Server/Controllers/UserProfileController.cs]: empty row

File: CryptoWorld.News/CryptoWorld.News.Server/appsettings.json [hunk 5]
```diff
@@ -12,6 +12,10 @@
   "ConnectionStrings": {
     "DefaultConnection": "Server=.;Database=CryptoWorldNews;Trusted_Connection=True;MultipleActiveResultSets=true;TrustServerCertificate=True"
   },
+
```
Comment (ZdravkoBorisov) [inline CryptoWorld.News/CryptoWorld.News.Server/appsettings.json]: Please exclude this from appsettings.json

File: CryptoWorld.News/cryptoworld.news.client/src/Components/Navbar.jsx [hunk 6]
```diff
@@ -2,6 +2,7 @@ import React from "react";
 import { Nav, NavLink, NavMenu } from "./NavbarElements";
 import Profile from "../Pages/ProfileDropdown.jsx";
 
+
```
Comment (ZdravkoBorisov) [inline CryptoWorld.News/cryptoworld.news.client/src/Components/Navbar.jsx]: Please remove extra line

File: CryptoWorld.News/cryptoworld.news.client/src/App.jsx [hunk 7]
```diff
@@ -5,6 +5,7 @@ import Navbar from './Components/Navbar.jsx';
 import Home from './Pages/Home';
 import Login from './Pages/Login';
 import Register from './Pages/Register';
+
```
Comment (ZdravkoBorisov) [inline CryptoWorld.News/cryptoworld.news.client/src/App.jsx]: Please remove extra line

File: CryptoWorld.News/CryptoWorld.News.Core/Services/UserProfileService.cs [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+            context = _context;
+        }
+        public async Task<ApplicationUser> EditProfileAsync(UserProfileModel model)
+        {
+            var user = await userManager.FindByIdAsync(model.Id);
+
+            if (user == null)
+                throw new ArgumentException("There is no such user.");
+            
+            user.FirstName = model.FirstName;
```
Comment (ZdravkoBorisov) [inline CryptoWorld.News/CryptoWorld.News.Core/Services/UserProfileService.cs:40]: Please add model validation. For example string.Empty or whitespace etc.

File: CryptoWorld.News/CryptoWorld.News.Core/Services/AccountService.cs [hunk 9]
```diff
@@ -16,16 +20,26 @@ namespace CryptoWorld.News.Core.Services
...
-            UserManager<ApplicationUser> _userManager, SignInManager<ApplicationUser> _signInManager, IEmailSenderService _emailSenderService)
+                UserManager<ApplicationUser> _userManager,
+                IConfiguration _config,
+                SignInManager<ApplicationUser> _signInManager,
+                IEmailSenderService _emailSenderService
+            )
         {
             userManager = _userManager;
+            config = _config;
+            this.secretKey = config["JWT:secretKey"];
```
Comment (ZdravkoBorisov) [inline CryptoWorld.News/CryptoWorld.News.Core/Services/AccountService.cs]: Get the JWT key from correct source

**Review by ZdravkoBorisov — CHANGES_REQUESTED** [review]

**Review by VelinArbov — CHANGES_REQUESTED** [review]

File: CryptoWorld.News/CryptoWorld.News.Server/Controllers/UserProfileController.cs [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+namespace CryptoWorld.Application.Server.Controllers
+{
+    [Authorize]
+    public class UserProfileController : BaseApiController
+    {
+        private readonly IUserProfileService profileService;
+        private readonly ApplicationDbContext context;
+        private readonly UserManager<ApplicationUser> userManager;
+        public UserProfileController(IUserProfileService _profileService, ApplicationDbContext _context, UserManager<ApplicationUser> _userManager)
```
Comment (VelinArbov) [inline CryptoWorld.News/CryptoWorld.News.Server/Controllers/UserProfileController.cs]: private field should be with _ like ---->   private readonly UserManager<ApplicationUser> _userManager; not in the constructer. Please fix whole PR

**Review by VelinArbov — CHANGES_REQUESTED** [review]

File: CryptoWorld.News/CryptoWorld.News.Server/Controllers/UserProfileController.cs (hunk 10 above)
Comment (VelinArbov) [reply CryptoWorld.News/CryptoWorld.News.Server/Controllers/UserProfileController.cs]: I mean every private fields

**Review by ZdravkoBorisov — APPROVED** [review]

**Pull request merged**